src/language-js/parser/try-combinations.js

```diff
@@ -0,0 +1,17 @@
+"use strict";
+
+function tryCombinations(combinations, fn) {
+  let result;
+  const errors = [];
```

Comment:
Does this need to be an array? If we only use the first one, I don't think it needs to be an array.

```js
function tryCombinations(combinations, fn) {
  let result;
  let error;
  for (const item of combinations) {
    try {
      result = fn(item);
      break;
    } catch (e) {
      if (!error) {
        error = e;
      }
    }
  }
  return [result, error];
}
```

Reply:
I thought babel throws the last error, I was going to use

```js
const [ast, ...errors] = tryCombinations
if(!ast) {
  throw getLast(errors)
}
```

but seems throw the first one passes the tests.

I think should be able to get all errors?

Reply:
I think there is no problem in the case that the first one is thrown.

Reply:
Updated to return only first error https://github.com/prettier/prettier/pull/9783/commits/21a6183f400a087edfd0bfec71d0cfe3da5d5cb5